src/editors/selectEditor.js

```diff
@@ -15,73 +15,6 @@ import { KEY_CODES } from './../helpers/unicode';
 import BaseEditor, { EditorState } from './_baseEditor';
 import { objectEach } from '../helpers/object';
 
-const SelectEditor = BaseEditor.prototype.extend();
-
-/**
- * @private
- * @editor SelectEditor
- * @class SelectEditor
- */
-SelectEditor.prototype.init = function() {
-  this.select = document.createElement('SELECT');
-  addClass(this.select, 'htSelectEditor');
-  this.select.style.display = 'none';
-  this.instance.rootElement.appendChild(this.select);
-  this.registerHooks();
-};
-
-SelectEditor.prototype.registerHooks = function() {
-  this.instance.addHook('afterScrollHorizontally', () => this.refreshDimensions());
-  this.instance.addHook('afterScrollVertically', () => this.refreshDimensions());
-  this.instance.addHook('afterColumnResize', () => this.refreshDimensions());
-  this.instance.addHook('afterRowResize', () => this.refreshDimensions());
-};
-
-SelectEditor.prototype.prepare = function(...args) {
-  BaseEditor.prototype.prepare.apply(this, args);
-
-  const selectOptions = this.cellProperties.selectOptions;
-  let options;
-
-  if (typeof selectOptions === 'function') {
-    options = this.prepareOptions(selectOptions(this.row, this.col, this.prop));
-  } else {
-    options = this.prepareOptions(selectOptions);
-  }
-
-  empty(this.select);
-
-  objectEach(options, (value, key) => {
-    const optionElement = document.createElement('OPTION');
-    optionElement.value = key;
-    fastInnerHTML(optionElement, value);
-    this.select.appendChild(optionElement);
-  });
-};
-
-SelectEditor.prototype.prepareOptions = function(optionsToPrepare) {
-  let preparedOptions = {};
-
-  if (Array.isArray(optionsToPrepare)) {
-    for (let i = 0, len = optionsToPrepare.length; i < len; i++) {
-      preparedOptions[optionsToPrepare[i]] = optionsToPrepare[i];
-    }
-  } else if (typeof optionsToPrepare === 'object') {
-    preparedOptions = optionsToPrepare;
-  }
-
-  return preparedOptions;
-
-};
-
-SelectEditor.prototype.getValue = function() {
-  return this.select.value;
-};
-
-SelectEditor.prototype.setValue = function(value) {
-  this.select.value = value;
-};
-
 const onBeforeKeyDown = function(event) {
```

Comment:
`onBeforeKeyDown` could be pulled into the editor class, similarly to [this example in the copyPaste plugin](https://github.com/handsontable/handsontable/blob/develop/src/plugins/copyPaste/copyPaste.js#L132).